torchqdynamics/odeint.py

```diff
@@ -90,7 +90,7 @@ def _fixed_odeint(
         y_save = torch.zeros(len(t_save), *y0.shape).to(y0)
 
     if len(exp_ops) > 0:
-        exp_save = torch.zeros(len(exp_ops), len(t_save)).to(y0)
+        exp_save = torch.zeros(*y0.shape[:-2], len(exp_ops), len(t_save)).to(y0)
```

Comment:
Following my comments on slack, I would argue that `y0` should always be of shape `(b,n,m)` inside `odeint`, whether it is called from `mesolve` or `ssolve`, and whether `b==1` or `m==1` or not. If we do that, we should replace this with `y0.size(0)` for readability. 

Also, I agree with this ordering I think. It's a bit more natural to me.

Reply:
I agree with you, cf. my reply on Slack. I suggest keeping this for this PR because the purpose is not to define the batching of `mesolve` arguments properly, rather to fix a bug and propose a perf improvement. I think setting the batching we want needs a little more discussion, so we all agree, and a proper separate PR.